File: lesson7giang/src/main/java/com/example/lesson7giang/StudentDAO.java
package com.example.lesson7giang;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface StudentDAO {

    @Query("SELECT * from student_table ORDER BY studentName ASC")
    LiveData<List<Student>> getAllStudent();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Student student);

    @Query("DELETE FROM student_table")
    void deleteAll();

}

File: lesson7khanh/src/main/java/com/k001/lesson7khanh/StudentDao.kt
package com.k001.lesson7khanh

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface StudentDao {

    @Query("SELECT * from student_table ORDER BY student_name ASC")
    fun getAllStudent() : LiveData<List<Student>>

    @Insert
    suspend fun insert(student: Student)

    @Delete
    fun deleteStudent(vararg student: Student)

    @Query("DELETE FROM student_table")
    fun deleteAll()
}